sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,164 +1,122 @@
-```
-
-### Deserialization
-
-```C# Snippet:ExplicitCast_Deserialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-Response response = client.GetPet("myPet");
-DogListProperty dog = (DogListProperty)response;
-Console.WriteLine(dog.IsHungry);
-```
+## Using the ModelSerializer
+The `ModelSerializer` class enables users to serialize and deserialize any Azure models that implement the `IModelSerializable<T>` interface. The following example shows a Dog model being serialized and deserialized.
 
-Given that explicit cast does not allow for serialization options we might also consider a static `FromResponse` and instance `ToRequestContent` methods.
+```C# Snippet:BaseModelSerializer
+Dog doggo = new Dog
+{
+    Name = "Doggo",
+    Age = 7
+};
 
-## Using ModelSerializer for NewtonSoftJson
+BinaryData data = ModelSerializer.Serialize(doggo);
 
-By using the ModelSerializer class, a new instance of Dog does not need to be created before calling Deserialize. Also added ObjectSerializer to Options class so different kinds of Serializers can be used.
+Dog dog = ModelSerializer.Deserialize<Dog>(data);
+```
 
-### Serialization
+By default, the ModelSerializer gives you the Json representation of the model. Some services accept XML so if you want to send the serialized data to an Azure service that accepts XML, you can use the `ModelSerializerFormat` enum to specify the format. By using the `Wire` format, the serializer automatically picks what the service would use.
 
-```C# Snippet:NewtonSoft_Serialize
-DogListProperty dog = new DogListProperty
+```C# Snippet:ModelSerializerWithFormat
+Dog doggo = new Dog
 {
     Name = "Doggo",
-    IsHungry = true,
-    Weight = 1.1,
-    FoodConsumed = { "kibble", "egg", "peanut butter" },
+    Age = 7
 };
-ModelSerializerOptions options = new ModelSerializerOptions();
-options.GenericTypeSerializerCreator = type => type.Equals(typeof(DogListProperty)) ? new NewtonsoftJsonObjectSerializer() : null;
-
-BinaryData data = ModelSerializer.Serialize(dog, options);
-```
 
-### Deserialization
+ModelSerializerOptions options = new ModelSerializerOptions(format: ModelSerializerFormat.Wire);
 
-```C# Snippet:NewtonSoft_Deserialize
-ModelSerializerOptions options = new ModelSerializerOptions();
-options.GenericTypeSerializerCreator = type => type.Equals(typeof(DogListProperty)) ? new NewtonsoftJsonObjectSerializer() : null;
-string json = @"[{""LatinName"":""Animalia"",""Weight"":1.1,""Name"":""Doggo"",""IsHungry"":false,""FoodConsumed"":[""kibble"",""egg"",""peanut butter""],""NumberOfLegs"":4}]";
+BinaryData data = ModelSerializer.Serialize(doggo, options);
 
-DogListProperty dog = ModelSerializer.Deserialize<DogListProperty>(BinaryData.FromString(json), options);
+Dog dog = ModelSerializer.Deserialize<Dog>(data, options);
 ```
 
-## Using ModelJsonConverter for JsonSerializer
+## Using models with protocol methods
```

Comment:
To clarify intent here - I recognize that we've chosen to call these "protocol methods" across languages and there is other documentation that discusses them in detail.

What I'm asking for here is just a sentence or maybe two that describes the term for the layperson so that they don't have to jump to another document just to get a definition.  We'll want to keep the link for folks that would like to dig deeper and understand the wider context.

Reply:
You can always hyperlink to the [protocol method](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md) doc as well.